src/nodiff.js

```diff
@@ -1,3 +1,126 @@
-export default async function nodiff() {
-  // TODO(dabrady) Write the code that does the thing.
+import { info, isDebug, setFailed } from '@actions/core';
+import { exec } from '@actions/exec';
+import {
+  hydrateTemplateString,
+  leaveComment,
+  requestChanges,
+  requestReviewers,
+} from './helpers';
+
+const FAILURE_MESSAGE = `Meaningless changes have been made to:\n`;
+
+/**
+ * This action lets you react when no meaningful changes are made within a given change set.
+ *
+ * Currently, a 'meaningless' change is defined in terms of whitespace, but it can be extended later. It can be
+ * configured to monitor an entire project or a set of files within it, and by default it simply fails if the given
+ * change set makes 'no difference' to the codebase.
+ *
+ * You can also configure it to react with one or more of a small set of predefined and slightly configurable actions,
+ * such as requesting a review from someone, leaving a comment, or requesting changes.
+ */
+export default async function nodiff({
+  filesToJudge,
+  baseGitRef,
+  doThisInResponse: {
+    requestReviewers: githubHandles,
+    comment,
+    fail
+  },
+  actionPayload,
+  githubToken
+}) {
+  // NOTE(dabrady) This prevents consumers from needing to do this themselves.
+  // We need to fetch the base to do a proper diff.
+  await fetchGitRef(baseGitRef);
+
+  // Get the list of files that have been changed meaninglessly.
+  var fileList = await meaninglessDiff(filesToJudge, baseGitRef);
```

Comment:
can't we just say `files`?

Reply:
I did originally, but it shadowed a different variable, so I renamed it to `filesList`. My later cleanup might have made that problem go away though, I'll try again 👍🏻

Reply:
Done 🎉